include/mbgl/util/string.hpp

```diff
@@ -10,7 +10,16 @@ namespace util {
 
 template <class T>
 inline std::string toString(T t) {
-    return std::to_string(t);
+    if (t - (int)t > 0.0 || t + (int)t < 0.0) {
```

Comment:
Yeah, it makes much more sense to have `toString(double num)`. I did this in e9d16dc07f5b1380ea6e66601548007860572bd4 and it seems to be working at least as well.

`long double` is not specifically handled ([see this test line](https://github.com/mapbox/mapbox-gl-native/pull/4965/files#diff-cb902609ef74233feb733fddb90c8b86R48)), but I think that’s an edge case we’re OK not covering.